contracts/EntryPoint.sol

```diff
@@ -319,34 +311,38 @@ contract EntryPoint is StakeManager {
     // create the sender's contract if needed.
     function _createSenderIfNeeded(MemoryUserOp memory mUserOp, bytes calldata initCode) internal {
         if (initCode.length != 0) {
-            // note that we're still under the gas limit of validate, so probably
-            // this create2 creates a proxy account.
-            // @dev initCode must be unique (e.g. contains the signer address), to make sure
-            //   it can only be executed from the entryPoint, and called with its initialization code (callData)
-            address sender1 = ICreate2Deployer(create2factory).deploy(initCode, bytes32(mUserOp.nonce));
-            require(sender1 != address(0), "create2 failed");
-            require(sender1 == mUserOp.sender, "sender doesn't match create2 address");
+            require(mUserOp.sender.code.length == 0, "sender already constructed");
+            address sender1 = _createSender(initCode);
+            require(sender1 == mUserOp.sender, "sender doesn't match initCode address");
+            require(sender1.code.length != 0, "initCode failed to create sender");
         }
     }
 
     /**
-     * Get counterfactual sender address.
-     *  Calculate the sender contract address that will be generated by the initCode and salt in the UserOperation.
-     * @param initCode the constructor code to be passed into the UserOperation.
-     * @param salt the salt parameter, to be passed as "nonce" in the UserOperation.
+     * call the "initCode" factory to create and return the sender wallet address
+     * initCode must be unique (e.g. contains the signer address), to make sure
+     *  it can only be executed from the entryPoint, and called with its initialization code (callData)
+     * @param initCode the initCode value from a UserOp. contains 20 bytes of factory address, followed by calldata
+     * @return sender the returned address of the created wallet.
      */
-    function getSenderAddress(bytes memory initCode, uint256 salt) public view returns (address) {
-        bytes32 hash = keccak256(
-            abi.encodePacked(
-                bytes1(0xff),
-                address(create2factory),
-                salt,
-                keccak256(initCode)
-            )
-        );
+    function _createSender(bytes calldata initCode) internal returns (address sender) {
+        address initAddress = address(bytes20(initCode[0 : 20]));
+        bytes memory initCallData = initCode[20 :];
+        bool success;
+        assembly {
```

Comment:
Does this have to be in assembly? Can't it be `(success, ret) = initAddress.call(initCallData)` kind of situation?